commit a148102e8ed56c7a8d0f790fad2f9366b27882d4
Author: Hrvoje Senjan <hrvoje.senjan@gmail.com>
Date:   Thu Feb 26 15:57:22 2015 +0100

    Register KSysguard to DBus
    
    Unbreaks starting the app via KRun
    
    Reviewed-by: Bhushan Shah

diff --git a/gui/CMakeLists.txt b/gui/CMakeLists.txt
index f021f0b..3f0ce93 100644
--- a/gui/CMakeLists.txt
+++ b/gui/CMakeLists.txt
@@ -54,6 +54,7 @@ target_compile_definitions(kdeinit_ksysguard PRIVATE -DPROJECT_VERSION="${PROJEC
 target_link_libraries(kdeinit_ksysguard
     KF5::ProcessUi
     KF5::SignalPlotter
+    KF5::DBusAddons
     KF5::ItemViews
     KF5::NewStuff
     KF5::SysGuard
diff --git a/gui/ksysguard.cpp b/gui/ksysguard.cpp
index 919fd89..fa77506 100644
--- a/gui/ksysguard.cpp
+++ b/gui/ksysguard.cpp
@@ -37,7 +37,7 @@
 #include <k4aboutdata.h>
 #include <kactioncollection.h>
 #include <kapplication.h>
-
+#include <KDBusService>
 #include <kdebug.h>
 #include <kedittoolbar.h>
 #include <kglobal.h>
@@ -533,6 +533,7 @@ extern "C" Q_DECL_EXPORT int kdemain( int argc, char** argv )
   KAboutData aboutData( "ksysguard", i18n( "System Monitor" ),
                         PROJECT_VERSION, i18n(Description), KAboutLicense::GPL,
                         i18n( "(c) 1996-2008 The KDE System Monitor Developers" ) );
+  aboutData.setOrganizationDomain(QByteArray("kde.org"));
   aboutData.addAuthor( i18n("John Tapsell"), i18n("Current Maintainer"), "john.tapsell@kde.org" );
   aboutData.addAuthor( i18n("Chris Schlaeger"), i18n("Previous Maintainer"), "cs@kde.org" );
   aboutData.addAuthor( i18n("Greg Martyn"), QString(), "greg.martyn@gmail.com" );
@@ -585,6 +586,9 @@ extern "C" Q_DECL_EXPORT int kdemain( int argc, char** argv )
   Toplevel->show();
   KSGRD::SensorMgr->setBroadcaster( Toplevel );  // SensorMgr uses a QPointer for toplevel, so it is okay if Toplevel is deleted first
 
+  // register to DBus
+  const KDBusService dbusService(KDBusService::Multiple);
+
   // run the application
   int result = app.exec();
 
